manifests/administration.pp

```diff
@@ -32,14 +33,21 @@
   }
 
   if $disable_thp {
-    exec { 'Disable Hugepages':
-      command => 'echo never > /sys/kernel/mm/transparent_hugepage/enabled',
-      path    => ['/sbin', '/usr/sbin', '/bin', '/usr/bin'],
-      onlyif  => 'test -f /sys/kernel/mm/transparent_hugepage/enabled',
-      unless  => 'cat /sys/kernel/mm/transparent_hugepage/enabled | grep "\[never\]"',
+    package { $hugeadm_package:
+      ensure => 'present',
     }
   }
 
+  $ensure_thp = $disable_thp ? { true => 'present', false => 'absent' }
+
+  systemd::timer { 'disable_thp.timer':
```

Comment:
the systemd::timer resource also needs to go into the `if $disable_thp {}` block

Reply:
Alongside the `$ensure_thp` definition ofc